test/hotspot/gtest/nmt/test_vmatree.cpp

```diff
@@ -739,4 +754,294 @@ TEST_VM_F(NMTVMATreeTest, SummaryAccountingWhenUseFlagInplace) {
+    //        Rl        Rs           C       Rs        Rl
+    //        -         si_1         si_1    si_1      -
+    //        -         -            si_2     -        -
+    ExpectedTree<4> et2 = {{     0,     20,     40,    100        },
+                           {mtNone, mtTest, mtTest, mtTest, mtNone},
+                           {Rl    , Rs    , C     , Rs    , Rl    },
+                           {-1    , si_1  , si_1  , si_1  , -1    },
+                           {-1    , -1    , si_2  , -1    , -1    }};
+    check_tree(tree, et2, __LINE__);
+
+    SIndex si_3 = si[2];
+    VMATree::RegionData call_stack_3(si_3, mtTest);
+    // commit with overlap at the region's start
+    tree.commit_mapping(10, 20, call_stack_3);
+    // Pre:  .........0-------------20**************40-------100.........
+    // Post: .........0---------10********30********40-------100.........
+    //        mtNone    mtTest     mtTest    mtTest    mtTest    mtNone
+    //        Rl        Rs           C       C         Rs        Rl
+    //        -         si_1         si_1    si_1      si_1      -
+    //        -         -            si_3    si_2        -       -
+    ExpectedTree<5> et3 = {{     0,     10,     30,     40,    100        },
+                           {mtNone, mtTest, mtTest, mtTest, mtTest, mtNone},
+                           {Rl    , Rs    , C     , C     , Rs    , Rl    },
+                           {-1    , si_1  , si_1  , si_1  , si_1  , -1    },
+                           {-1    , -1    , si_3  , si_2  , -1    , -1    }};
+    check_tree(tree, et3, __LINE__);
+
+    SIndex si_4 = si[3];
+    VMATree::RegionData call_stack_4(si_4, mtTest);
+    // commit with overlap at the region's end
+    tree.commit_mapping(30, 20, call_stack_4);
+    // Pre:  .........0---------10********30**40-------------100.........
+    // Post: .........0---------10********30********50-------100.........
+    //        mtNone    mtTest     mtTest    mtTest    mtTest    mtNone
+    //        Rl        Rs           C       C         Rs        Rl
+    //        -         si_1         si_1    si_1      si_1      -
+    //        -         -            si_3    si_4        -       -
+    ExpectedTree<5> et4 = {{     0,     10,     30,     50,    100        },
+                           {mtNone, mtTest, mtTest, mtTest, mtTest, mtNone},
+                           {Rl    , Rs    , C     , C     , Rs    , Rl    },
+                           {-1    , si_1  , si_1  , si_1  , si_1  , -1    },
+                           {-1    , -1    , si_3  , si_4  , -1    , -1    }};
+    check_tree(tree, et4, __LINE__);
+  }
+  {// uncommit should not store any call-stack
+    Tree tree;
+    tree.reserve_mapping(0, 100, call_stack_1);
+
+    tree.commit_mapping(20, 20, call_stack_2, true);
+
+    tree.commit_mapping(0, 10, call_stack_2, true);
+
+    tree.uncommit_mapping(0, 5, call_stack_2);
+    // Pre:  .........0********10--------20********40-------100.........
```

Comment:
I think this one got lost, still could use reformatting:

`// Pre:  .........0*****************10--------20********40-------100.........`